tests/unit/python/foglamp/services/core/api/test_scheduler_api.py

```diff
@@ -0,0 +1,211 @@
+                    assert 'purge' == json_response
+                mock_storage_call.assert_called_with('scheduled_processes', payload)
+
+    async def test_get_scheduled_process_bad_data(self, client):
+        storage_client_mock = MagicMock(StorageClient)
+        response = {'rows': [], 'count': 0}
+        with patch.object(connect, 'get_storage', return_value=storage_client_mock):
+                with patch.object(storage_client_mock, 'query_tbl_with_payload',
+                                  return_value=response):
+                    resp = await client.get('/foglamp/schedule/process/bla')
+                    assert 404 == resp.status
+                    assert 'No such Scheduled Process: bla.' == resp.reason
+
+
+class TestSchedules:
+    _random_uuid = uuid.uuid4()
+
+    @pytest.fixture
+    def client(self, loop, test_client):
+        app = web.Application(loop=loop)
+        # fill the routes table
+        routes.setup(app)
+        return loop.run_until_complete(test_client(app))
+
+    async def test_get_schedules(self, client):
+        async def mock_coro():
+            schedules = []
+            schedule = StartUpSchedule()
+            schedule.schedule_id = "1"
+            schedule.exclusive = True
+            schedule.enabled = True
+            schedule.name = "foo"
+            schedule.process_name = "bar"
+            schedule.repeat = timedelta(seconds=30)
+            schedule.time = None
+            schedule.day = None
+            schedules.append(schedule)
+            return schedules
+
+        server.Server.scheduler = Scheduler(None, None)
+        with patch.object(server.Server.scheduler, 'get_schedules', return_value=mock_coro()):
+            resp = await client.get('/foglamp/schedule')
+            assert 200 == resp.status
+            result = await resp.text()
+            json_response = json.loads(result)
+            assert {'schedules': [
+                {'name': 'foo', 'day': None, 'type': 'STARTUP', 'processName': 'bar',
+                 'time': 0, 'id': '1', 'exclusive': True, 'enabled': True, 'repeat': 30.0}
+            ]} == json_response
+
+    async def test_get_schedule(self, client):
+        async def mock_coro():
+            schedule = StartUpSchedule()
+            schedule.schedule_id = self._random_uuid
+            schedule.exclusive = True
+            schedule.enabled = True
+            schedule.name = "foo"
+            schedule.process_name = "bar"
+            schedule.repeat = timedelta(seconds=30)
+            schedule.time = None
+            schedule.day = None
+            return schedule
+
+        server.Server.scheduler = Scheduler(None, None)
+        with patch.object(server.Server.scheduler, 'get_schedule', return_value=mock_coro()):
+            resp = await client.get('/foglamp/schedule/{}'.format(self._random_uuid))
+            assert 200 == resp.status
+            result = await resp.text()
+            json_response = json.loads(result)
```

Comment:
result is actual json_response, you mean response_dict = json.loads(result) ?

Reply:
```
print(type(result))
 <class 'str'>
```